### PR TITLE
Rip bazel out

### PR DESCRIPTION
**Pull request by jonjohnsonjr**
I cannot understand why, but running the ko tests breaks my editor. When
it tries to run goimports on save, it breaks iff I run the ko unit
tests. Every time. I think the build package is *not* safe :)

Output:

```
$ go test ./pkg/... -count=1
ok  	github.com/google/go-containerregistry/pkg/authn	0.034s
?   	github.com/google/go-containerregistry/pkg/crane	[no test files]
?   	github.com/google/go-containerregistry/pkg/ko	[no test files]
ok  	github.com/google/go-containerregistry/pkg/ko/build	0.031s
open /tmp/go-build243642577/b184/vet.cfg: no such file or directory
go build github.com/google/go-containerregistry/pkg/v1 (testmain): open /tmp/go-build243642577/b185/importcfg: no such file or directory
go build github.com/google/go-containerregistry/pkg/v1/daemon (testmain): open /tmp/go-build243642577/b188/importcfg: no such file or directory
open /tmp/go-build243642577/b182/name.test: no such file or directory
go build github.com/google/go-containerregistry/pkg/v1/empty (testmain): open /tmp/go-build243642577/b191/importcfg: no such file or directory
go build github.com/google/go-containerregistry/pkg/v1/mutate (testmain): open /tmp/go-build243642577/b194/importcfg: no such file or directory
open /tmp/go-build243642577/b077/vet.cfg: no such file or directory
go build github.com/google/go-containerregistry/pkg/v1/random (testmain): open /tmp/go-build243642577/b197/importcfg: no such file or directory
go build github.com/google/go-containerregistry/pkg/v1/remote (testmain): open /tmp/go-build243642577/b200/importcfg: no such file or directory
open /tmp/go-build243642577/b137/publish.test: no such file or directory
FAIL	github.com/google/go-containerregistry/pkg/ko/publish [build failed]
go build github.com/google/go-containerregistry/pkg/v1/remote/transport (testmain): open /tmp/go-build243642577/b203/importcfg: no such file or directory
go build github.com/google/go-containerregistry/pkg/v1/tarball (testmain): open /tmp/go-build243642577/b206/importcfg: no such file or directory
open /tmp/go-build243642577/b062/vet.cfg: no such file or directory
go build github.com/google/go-containerregistry/pkg/v1/v1util (testmain): open /tmp/go-build243642577/b209/importcfg: no such file or directory
open /tmp/go-build243642577/b170/resolve.test: no such file or directory
FAIL	github.com/google/go-containerregistry/pkg/ko/resolve [build failed]
FAIL	github.com/google/go-containerregistry/pkg/name [build failed]
FAIL	github.com/google/go-containerregistry/pkg/v1 [build failed]
FAIL	github.com/google/go-containerregistry/pkg/v1/daemon [build failed]
FAIL	github.com/google/go-containerregistry/pkg/v1/empty [build failed]
FAIL	github.com/google/go-containerregistry/pkg/v1/mutate [build failed]
FAIL	github.com/google/go-containerregistry/pkg/v1/random [build failed]
FAIL	github.com/google/go-containerregistry/pkg/v1/remote [build failed]
FAIL	github.com/google/go-containerregistry/pkg/v1/remote/transport [build failed]
FAIL	github.com/google/go-containerregistry/pkg/v1/tarball [build failed]
FAIL	github.com/google/go-containerregistry/pkg/v1/v1util [build failed]

$ go test ./pkg/v1/... -count=1
ok  	github.com/google/go-containerregistry/pkg/v1	0.028s
ok  	github.com/google/go-containerregistry/pkg/v1/daemon	0.057s
ok  	github.com/google/go-containerregistry/pkg/v1/empty	0.028s
ok  	github.com/google/go-containerregistry/pkg/v1/mutate	0.064s
?   	github.com/google/go-containerregistry/pkg/v1/partial	[no test files]
ok  	github.com/google/go-containerregistry/pkg/v1/random	0.053s
ok  	github.com/google/go-containerregistry/pkg/v1/remote	0.079s
ok  	github.com/google/go-containerregistry/pkg/v1/remote/transport	0.014s
ok  	github.com/google/go-containerregistry/pkg/v1/tarball	0.066s
?   	github.com/google/go-containerregistry/pkg/v1/types	[no test files]
ok  	github.com/google/go-containerregistry/pkg/v1/v1util	0.003s


$ go test ./pkg/ko/... -count=1
?   	github.com/google/go-containerregistry/pkg/ko	[no test files]
ok  	github.com/google/go-containerregistry/pkg/ko/build	0.045s
open /tmp/go-build372538965/b087/publish.test: no such file or directory
FAIL	github.com/google/go-containerregistry/pkg/ko/publish [build failed]
open /tmp/go-build372538965/b163/resolve.test: no such file or directory
FAIL	github.com/google/go-containerregistry/pkg/ko/resolve [build failed]

````

Fixes: https://github.com/google/go-containerregistry/issues/198

### CODE REVIEW

**Comment by jonjohnsonjr** [comment]
Setting GOTMPDIR seems to fix things!

```
$ GOTMPDIR=~/gotmp/ go test ./pkg/... -count=1
ok  	github.com/google/go-containerregistry/pkg/authn	0.011s
?   	github.com/google/go-containerregistry/pkg/crane	[no test files]
?   	github.com/google/go-containerregistry/pkg/ko	[no test files]
ok  	github.com/google/go-containerregistry/pkg/ko/build	0.013s
ok  	github.com/google/go-containerregistry/pkg/ko/publish	0.016s
ok  	github.com/google/go-containerregistry/pkg/ko/resolve	3.935s
ok  	github.com/google/go-containerregistry/pkg/name	0.021s
ok  	github.com/google/go-containerregistry/pkg/v1	0.002s
ok  	github.com/google/go-containerregistry/pkg/v1/daemon	0.026s
ok  	github.com/google/go-containerregistry/pkg/v1/empty	0.014s
ok  	github.com/google/go-containerregistry/pkg/v1/mutate	0.090s
?   	github.com/google/go-containerregistry/pkg/v1/partial	[no test files]
ok  	github.com/google/go-containerregistry/pkg/v1/random	0.063s
ok  	github.com/google/go-containerregistry/pkg/v1/remote	0.083s
ok  	github.com/google/go-containerregistry/pkg/v1/remote/transport	0.035s
ok  	github.com/google/go-containerregistry/pkg/v1/tarball	0.104s
?   	github.com/google/go-containerregistry/pkg/v1/types	[no test files]
ok  	github.com/google/go-containerregistry/pkg/v1/v1util	0.003s
```

I do not yet know how to turn this new found knowledge into a fix!

**Comment by jonjohnsonjr** [comment]
@ImJasonH can you teach me what I need to do here?

https://github.com/google/go-containerregistry/blob/master/cmd/crane/rebase_cloudbuild.yaml#L2-L8

**Comment by mattmoor** [comment]
@jonjohnsonjr That can probably become `ko publish` ;-P

**Comment by ImJasonH** [comment]
`ko` doesn't currently like to be run outside of `GOPATH` which will be a bit of a roadblock to using it in GCB.

Luckily, un-Bazeling will help us make progress on that too. If rebase_cloudbuild.yaml is broken in the meantime that's probably okay.

**Comment by jonjohnsonjr** [comment]
> If rebase_cloudbuild.yaml is broken in the meantime that's probably okay.

SGTM 😄 

Ok @mattmoor this is RFAL!

File: .travis.yml [hunk 1]
```diff
@@ -1,4 +1,4 @@
-sudo: required
+sudo: false
```
Comment (mattmoor) [inline .travis.yml]: 🎉

File: pkg/ko/build/gobuild_test.go [hunk 2]
```diff
@@ -31,22 +31,26 @@ type testContext struct {
...
 	ogp := os.Getenv("GOPATH")
+
+	// Run the tests.
+	code := m.Run()
+
+	// Restore everything.
+	getwd = os.Getwd
+	os.Setenv("GOPATH", ogp)
+
+	os.Exit(code)
```
Comment (mattmoor) [inline pkg/ko/build/gobuild_test.go]: Are you sure this is still necessary?  EnvVars should be scoped to the process.

**Comment by jonjohnsonjr** [comment]
> FWIW, I'd have staged this as two changes

Ack, sorry :P

File: pkg/ko/build/gobuild_test.go (hunk 2 above)
Comment (jonjohnsonjr) [reply pkg/ko/build/gobuild_test.go]: Tests passed without it, gone!

**Comment by jonjohnsonjr** [comment]
Quite the difference:

image

**Comment by jonjohnsonjr** [comment]
Rebased!

**Pull request merged**